### ISSUE TITLE
Should be able to pass API keys as params rather than requiring them to be in the environment

### ISSUE THREAD

**Issue by camjw** [hwchase17/langchain#76]
It's just a bit annoying, I want to use this library in production and I currently store credentials not in the environment.

I think the ideal API is like all the AWS SDKs where you can either stick them in the environment OR pass them as params to the llm constructor.

I can do a PR for this if you're accepting PRs?

**Comment by hwchase17** [comment]
great idea. prs are def being accepted! will keep an eye out for it. at the same time, this is such a good idea im more than happy to work on myself if you dont have the time - lmk

**Comment by camjw** [comment]
Hmm so I just cloned the repo and I think the way you have it set up is that all of the executables (black, isort, mypy, etc) have to be installed in the system rather than any virtual environment. Would it be okay if I also switched to using poetry (https://python-poetry.org/) for dependency management? It's not perfect either, but it automatically sets up isolated venvs for you and the only difference is instead of running `pip install foo` you write `poetry add foo` etc and your global namespace doesn't get polluted.

No stress if not, realise this is quite bikesheddy!
